cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -480,10 +481,11 @@ NullOptions::NullOptions(bool nan_is_null)
     : FunctionOptions(internal::kNullOptionsType), nan_is_null(nan_is_null) {}
 constexpr char NullOptions::kTypeName[];
 
-PadOptions::PadOptions(int64_t width, std::string padding)
+PadOptions::PadOptions(int64_t width, std::string padding, bool align_left_on_odd_padding)
     : FunctionOptions(internal::kPadOptionsType),
       width(width),
-      padding(std::move(padding)) {}
+      padding(std::move(padding)),
+      align_left_on_odd_padding(align_left_on_odd_padding) {}
```

Comment:
You could call it `leftist` or `leftist_on_odd_padding`. No joke.

https://en.wikipedia.org/wiki/Leftist_tree

Reply:
While it might not be a joke ;), I would personally still not go for "leftist", because I think not many people would understand that without the wiki reference ...